src/display/annotation_layer.js

```diff
@@ -1062,26 +1108,27 @@ class RadioButtonWidgetAnnotationElement extends WidgetAnnotationElement {
     element.setAttribute("id", id);
     element.tabIndex = DEFAULT_TAB_INDEX;
 
-    element.addEventListener("change", function (event) {
-      const { target } = event;
-      for (const radio of document.getElementsByName(target.name)) {
-        if (radio !== target) {
-          storage.setValue(radio.getAttribute("id"), { value: false });
-        }
+    element.addEventListener("change", event => {
+      const { name, checked } = event.target;
+      for (const radio of this._getElementsByName(name, /* skipId = */ id)) {
+        storage.setValue(radio.id, { value: false });
       }
-      storage.setValue(id, { value: target.checked });
+      storage.setValue(id, { value: checked });
     });
 
     if (this.enableScripting && this.hasJSActions) {
       const pdfButtonValue = data.buttonValue;
+      const self = this;
       element.addEventListener("updatefromsandbox", jsEvent => {
         const actions = {
           value(event) {
             const checked = pdfButtonValue === event.detail.value;
-            for (const radio of document.getElementsByName(event.target.name)) {
-              const radioId = radio.getAttribute("id");
-              radio.checked = radioId === id && checked;
-              storage.setValue(radioId, { value: radio.checked });
+            for (const radio of self._getElementsByName(event.target.name)) {
```

Comment:
Replace `value(event) {` with `value: event => {` to avoid the need for capturing `self = this`.